app/_src/gateway/reference/expressions-language/language-references.md

```diff
@@ -0,0 +1,165 @@
+  the intention of the author is unclear.
+
+### IpAddr
+
+`IpCidr` represents a single IP addresses in [IPv4 Dot-decimal notation](https://en.wikipedia.org/wiki/Dot-decimal_notation),
+or the standard [IPv6 Address Format](https://en.wikipedia.org/wiki/IPv6_address#Address_formats).
+
+Here are some examples:
+
+IPv4:
+
+```
+net.src.ip == 192.168.1.1
+```
+
+IPv6:
+```
+net.src.ip == fd00::1
+```
+
+### Int
+
+There is only one integer type in Expressions, all integers are signed 64-bit integer. Integer
+literals can be written as: `12345`, `-12345`, or in hexadecimal format such as `0xab12ff`
+or in octet format as `0751`.
+
+### Regex
+
+Regex are written as `String` literals, but they are parsed when the regex operator `~` is present
+and checked for validity according to the [Rust `regex` crate syntax](https://docs.rs/regex/latest/regex/#syntax).
+For example, in the following predicate, the constant is parsed as a `Regex`:
+
+```
+http.path ~ r#"/foo/bar/.+"#
+```
+
+## Operators
+
+Expressions language support rich set of operators that can be performed on various data types.
+Here is an overview:
+
+| Operator        | Name                  | Description                                                                           |
+|-----------------|-----------------------|---------------------------------------------------------------------------------------|
+| `==`            | Equals                | Field value equals to constant value                                                  |
+| `!=`            | Not equals            | Field value does not equals to constant value                                         |
+| `~`             | Regex match           | Field value matches regex                                                             |
+| `^=`            | Prefix match          | Field value starts with constant value                                                |
+| `=^`            | Postfix match         | Field value ends with constant value                                                  |
+| `>=`            | Greater than or equal | Field value greater than or equal constant value                                      |
+| `>`             | Greater than          | Field value greater than constant value                                               |
+| `<=`            | Less than or equal    | Field value less than or equal constant value                                         |
+| `<`             | Less than             | Field value less than constant value                                                  |
+| `in`            | In                    | Field value is inside constant value                                                  |
+| `not in`        | Not in                | Field value is not inside constant value                                              |
+| `contains`      | Contains              | Field value contains constant value                                                   |
```

Comment:
Could you add some explanation on the difference between `in`/`not in` and `contains`?